### PR TITLE
updating french.pattern

### PR DESCRIPTION
**Pull request by cramoisi**
see a bunch of https://github.com/koreader/koreader/issues/3848 for context

### CODE REVIEW

File: cr3gui/data/hyph/French.pattern [hunk 1]
```diff
@@ -1,1255 +1,1887 @@
-<?xml version="1.0" encoding="utf8"?>
+<?xml version="1 0" encoding="utf8"?>
```
[the remark this answers is not in the document]
Comment (cramoisi) [reply cr3gui/data/hyph/French.pattern]: gosh, it's the only line with a dot appart from the pattern before converting. so the regexp took it and I didn't see it.

**Comment by Frenzie** [comment]
Hm, looks like there are more patterns with issues. I'm thinking of adding something like this to the CI to automatically catch issues like the one I identified above (because I only spotted it because it was right on top; I'm hardly going to read the whole XML file):

```
xmllint $(git ls-files "*.pattern") >/dev/null
```

The main problem is that there are a lot of double hyphens (`--`) within comments.

**Comment by Frenzie** [comment]
I guess they come from the TeX patterns:

https://github.com/koreader/crengine/blob/b3a261270596872ecb2c7deeaa9627c0cc84f286/cr3gui/data/hyph/French.pattern#L63-L69

https://github.com/koreader/crengine/blob/b3a261270596872ecb2c7deeaa9627c0cc84f286/cr3gui/data/hyph/French.pattern#L97-L113

**Comment by cramoisi** [comment]
huh ? I removed them. I send that : 
[hyph_fr.txt](https://github.com/koreader/crengine/files/1918823/hyph_fr.txt)

**Comment by Frenzie** [comment]
Yes, that's the current pattern. :-) In fact your pattern is perfect. XML-wise, anyway.

**Comment by cramoisi** [comment]
but the --- _where_ in the old files (I only add 4 commented sections and there is no --- inside them

**Comment by cramoisi** [comment]
Obviously, the .pattern converted before (not by me haha) would need a refresh for xml's sake. I can do it if you want.

**Comment by Frenzie** [comment]
You mean the Norwegian and Greek (possibly more)? That'd be neat! Meanwhile I'll implement the check to prevent bad XML from sneaking in.

**Comment by cramoisi** [comment]
yup. there also the german.pattern that need to be pull as it was tested this week-end.

**Pull request merged**